uk.co.compucorp.civicrm.hrcore/CRM/HRCore/CMSData/UserMailNotifier/Drupal.php

```diff
@@ -8,30 +8,25 @@
 class CRM_HRCore_CMSData_UserMailNotifier_Drupal implements UserMailNotifierInterface {
 
   /**
-   * @var stdClass
-   */
-  private $user;
-
-  /**
-   * CRM_HRCore_CMSData_UserMailNotifier_Drupal constructor.
+   *{@inheritdoc}
    *
-   * @param array $contactData
+   * @return \stdClass
    */
-  public function __construct($contactData) {
-    $this->user = user_load($contactData['cmsId']);
+  public function getUser($contactData) {
```

Comment:
Does this method really need to be public? What I suggested (sorry if I wasn't clear) was something more or less like this:

```php
public function sendWelcomeEmail($contactData) {
  $user = $this->getUser($contactData);
  return _user_mail_notify('register_admin_created', $user);
}
``` 

This way, `getUser()` will only be used internally, so there's no need for it to be public